### PR TITLE
fix(job): command exited with non zero reports to user

### PR DESCRIPTION
**Pull request by cedricbuild**
## Description

fix(job): command exited with non zero reports to user 

Fixes issue #

## TODOs:

- [ ] Changes to fish usage are reflected in user documentation/manpages.
- [ ] Tests have been added for regressions fixed
-  x] User-visible changes noted in CHANGELOG.rst

### CODE REVIEW

File: tests/pexpects/job_summary.py [hunk 1]
```diff
@@ -1,4 +1,6 @@
 #!/usr/bin/env python3
+import sys
+sys.path.append("/home/cedric/projects/fish-shell/build_tools/")
```
Comment (ridiculousfish) [inline tests/pexpects/job_summary.py:3]: Please remove these

File: share/functions/fish_job_summary.fish [hunk 2]
```diff
@@ -29,6 +29,8 @@ function fish_job_summary -a job_id is_foreground cmd_line signal_or_end_name si
 
     set -l message
     switch $signal_or_end_name
+        case STATUS
+            set message (printf ( _ "fish: Job %s, '%s' has Non Zero status code:%s\n" ) $job_id $cmd_line $signal_desc)
```
Comment (ridiculousfish) [inline share/functions/fish_job_summary.fish:33]: As `$signal_desc` may now be either the exit code or the status value, please update the "docstring" at the top of the function, and rename the variable.

As for the message, I suggest just extending the ENDED message:

```
set message (printf ( _ "fish: Job %s, '%s' has ended with status %d\n" ) $job_id $cmd_line $status_num)
```

File: src/proc.rs [hunk 3]
```diff
@@ -1628,12 +1661,17 @@ fn summary_command(j: &Job, p: Option<&Process>) -> WString {
         Some(p) => {
             // We are summarizing a process which exited with a signal.
```
Comment (ridiculousfish) [inline src/proc.rs:1662]: This comment is no longer accurate, please update it

**Review by ridiculousfish — CHANGES_REQUESTED** [review]
Thanks for this PR! Some things to fix.

**Comment by faho** [comment]
Closing this for lack of response.

**Pull request closed without merging**